common/source/docs/common-network.rst

```diff
@@ -0,0 +1,141 @@
+
+- A serial device supported by ArduPilot (ie. Rangefinder, Gimbal, serial ESC, etc.)
+- A MAVLink connection to another vehicle or GCS
+
+In addition, LUA Scripting running on an autopilot can access these data ports.
+
+.. image:: ../../../images/Net_framework.jpg
+
+Configuration
+=============
+
+- :ref:`NET_ENABLED<NET_ENABLED>` must be set to "1" and autopilot rebooted in order to see the network parameters:
+
+Ethernet MAC Configuration
+--------------------------
+
+If the autopilot has an Ethernet MAC the following parameters apply:
+
+- :ref:`NET_MACADDR0<NET_MACADDR0>`
+- :ref:`NET_MACADDR1<NET_MACADDR1>`
+- :ref:`NET_MACADDR2<NET_MACADDR2>`
+- :ref:`NET_MACADDR3<NET_MACADDR3>`
+- :ref:`NET_MACADDR4<NET_MACADDR4>`
+- :ref:`NET_MACADDR5<NET_MACADDR5>`
+
+This is the global 6 byte MAC address. The default is automatically generated from the autopilot's CPU unique id, but can be changed if desired.
+
+- :ref:`NET_IPADDR0<NET_IPADDR0>`
+- :ref:`NET_IPADDR1<NET_IPADDR1>`
+- :ref:`NET_IPADDR2<NET_IPADDR2>`
+- :ref:`NET_IPADDR3<NET_IPADDR3>`
+
+This is the autopilot's 4 byte IP address. It defaults to ``192.168.13.14``. It can be dynamically assigned if the network has a DHCP server by enabling ArduPilot's DHCP client using the :ref:`NET_DHCP<NET_DHCP>` parameter, or manually set as desired.
+
+- :ref:`NET_NETMASK<NET_NETMASK>` is the number of leading bits set in the subnet mask. For example, if its set to 24, then the subnet mask is  255.255.255.0, which is the default.
+- :ref:`NET_GWADDR0<NET_GWADDR0>`
+- :ref:`NET_GWADDR1<NET_GWADDR1>`
+- :ref:`NET_GWADDR2<NET_GWADDR2>`
+- :ref:`NET_GWADDR3<NET_GWADDR3>`
+
+This is the gateway's IP address out of the local sub-net defined by the :ref:`NET_NETMASK<NET_NETMASK>` mask. Any destination IPs that are outside the masked subnet range will be sent to the gateway's MAC address with the final destinations IP address.
+
+PPP Configuration
+-----------------
+
+This feature allows an H7 cpu-based autopilot to connect to networks via a Serial port using PPP. 
+
+.. note:: this cannot be used with autopilots having a built-in Ethernet MAC. Only one network connection per autopilot is allowed currently. Also, only one serial port on the autopilot can be configured for PPP.
+
+To enable this feature, it first must be present in the autopilot firmware. This can be done using the `Custom Firmware Build Server <https://firmware.ardupilot.org/>`__ or by building the code locally using the "--enable-ppp" waf configuration option (See :ref:`building-the-code`)
+
+To configure a serial port for PPP (Serial2 is used in this example):
+
+- set :ref:`SERIAL5_PROTOCOL<SERIAL5_PROTOCOL>` = 48 (PPP)
+- set :ref:`SERIAL5_BAUD<SERIAL5_BAUD>` = 12500000 (12.5MBaud)
```

Comment:
I've found that without flow control (I have tried on 2 different boards), the sweet spot seems to be 921,600 baud.

Reply:
added